src/core/server/logging/appenders/rewrite/policies/meta/meta_policy.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { LogRecord } from '@kbn/logging';
+import { set } from '@elastic/safer-lodash-set';
+import { has, unset } from 'lodash';
+import { assertNever } from '@kbn/std';
+import { RewritePolicy } from '../policy';
+
+const META_PATH = 'meta';
+
+export interface MetaRewritePolicyConfig {
+  type: 'meta';
+
+  /**
+   * The 'mode' specifies what action to perform on the
+   * specified properties.
+   *
+   * @remarks
+   *   - 'add' creates a new property at the provided 'path', skipping properties which already exist.
+   *   - 'update' updates an existing property at the provided 'path' without creating new properties.
+   *   - 'remove' removes an existing property at the provided 'path'.
+   */
+  mode: 'add' | 'remove' | 'update';
+
+  /**
+   * The properties to add, remove, or update.
+   *
+   * @remarks
+   * Each provided 'path' is relative to the record's {@link LogMeta}.
+   * For the 'remove' mode, no 'value' is provided.
+   */
+  properties: Array<{ path: string; value?: string | number | boolean | null }>;
+}
+
+export const metaRewritePolicyConfigSchema = schema.object({
+  type: schema.literal('meta'),
+  mode: schema.oneOf([schema.literal('add'), schema.literal('update'), schema.literal('remove')], {
+    defaultValue: 'update',
+  }),
+  properties: schema.arrayOf(
+    schema.object({
+      path: schema.string(),
+      value: schema.maybe(
+        schema.nullable(schema.oneOf([schema.string(), schema.number(), schema.boolean()]))
```

Comment:
What would happen if we used changed the value at a path to something other than the original type for that value?
e.g:
```
	censor:
      kind: rewrite
      appenders: [console, file]
      policy:
        type: meta
        mode: update
        properties:
          - path: "http.request.headers.cookie"
            value: true
```
It's a curiosity question more than a code comment

Reply:
> What would happen if we used changed the value at a path to something other than the original type for that value?

In the current implementation it would overwrite with whatever value you provided - at runtime there is no comparison between the existing value type with the replacement type.

Reply:
I thought so, thanks. I'm assuming this isn't a problem and that folks indexing logs and parsing them for whatever purpose would know what their doing and handle the type change on their side. It's not a concern for now, we'll reconsider the consequences down the line if/when we implement other policy types.

Reply:
> It's not a concern for now, we'll reconsider the consequences down the line if/when we implement other policy types.

Yeah I think it's still a valid point though. The main issue is that there isn't any way to know if the types are compatible until runtime, so if we wanted to enforce this, we'd either need to attempt to cast types, or throw an error.

Neither seems like a very appealing option. In particular throwing an error would be problematic as the person configuring the filter doesn't have control over the logger that's being filtered - so an error could be thrown any time after Kibana has already started, leading to unexpected server crashes.

For that reason, I agree with your assessment above that it is probably best to wait and see if this becomes a concern in practice. TBH I had originally implemented this just as a string, and only later added the other types as they are still valid yaml. But I'd be willing to bet the vast majority of filtering use cases are for string values.